### PR TITLE
Document non string types in template literal types

### PR DESCRIPTION
**Pull request by everett1992**
I was complaining that using `${number}` in template literal types was an undocumented feature so I did my best to document non string types in template literals to the best of my knowledge. 

microsoft/TypeScript#46674

### CODE REVIEW

File: packages/documentation/copy/en/handbook-v2/Type Manipulation/Template Literal Types.md [hunk 1]
```diff
@@ -216,3 +216,39 @@ function applyStringMapping(symbol: Symbol, str: string) {
...
+// These are all assignable to `${bigint}`
+bigint = '1' // decimal
+bigint = '0b10' // binary
+bigint = '0x10' // hexidecimal
+bigint = '-1' // negative numbers
+
+// These are errors because the runtime does not parse them to finite numbers.
+bigint = "NaN"
+bigint = "Infinity"
+bigint = "1000n" // 1000n is a literal BigInt but is not parsed as a BigInt.
```
Comment (everett1992) [inline packages/documentation/copy/en/handbook-v2/Type Manipulation/Template Literal Types.md]: I left it out of the examples but `${bigint}` _doesnt_ allow whitespace while `${number}` does, despite BigInt(' ') parsing to 0n. That breaks the original rule as I understood it. Is that a bug in typescript?

**Review by orta — COMMENTED** [review]
Yeah, I like this overall - you'll get some CI errors because twoslash requires you declare the known error messages in the code sample, get those in and it's 👍🏻  for me

**Comment by everett1992** [comment]
https://www.typescriptlang.org/dev/twoslash/ seems to error on any template literal type. 
I'll clone the project so I can run tests and view the rendered content.

File: packages/documentation/copy/en/handbook-v2/Type Manipulation/Template Literal Types.md [hunk 2]
```diff
@@ -216,3 +216,45 @@ function applyStringMapping(symbol: Symbol, str: string) {
...
 </details>
+
+### Non-string Types in Template Literals
+
+You can use the types `boolean`, `null`, `undefined`, `number`, and `bigint` in template literal types.
+`${boolean}` is equivelant to `"true" | "false"`, `${null}` and `${undefined}` match the strings `null` and `undefined` respectivly. 
+`${number}` and `${bigint}` match values that the JavaScript will parse to finite numbers. 
+
+```ts twoslash
+let number: `${number}`
```
Comment (orta) [inline packages/documentation/copy/en/handbook-v2/Type Manipulation/Template Literal Types.md]:
```suggestion
```ts twoslash
// @errors: 2322
let number: `${number}`
```

Comment (orta) [reply packages/documentation/copy/en/handbook-v2/Type Manipulation/Template Literal Types.md]: should do it I think

**Review by orta — APPROVED** [review]

**Comment by orta** [comment]
Looks good - just the CLA to go 👍🏻

**Comment by orta** [comment]
Ping on the CLA ^

**Comment by everett1992** [comment]
I'm waiting on approval from my employer

**Comment by everett1992** [comment]
CLA is signed, is anything blocking merge of this PR?

**Pull request closed without merging**